### ISSUE TITLE
Kallisto BUS fails to detect mitochondrial reads

### ISSUE THREAD

**Issue by chriscainx** [pachterlab/kallisto#212]
I'm running kallisto vs cellranger on PBMC8k v2.0 chemistry, and it seems that Kallisto doesn't report any mitochondria reads. What could be the reason?

![image](https://user-images.githubusercontent.com/8361080/58758548-5666d080-854f-11e9-82f2-9b8dfb599fae.png)
![image](https://user-images.githubusercontent.com/8361080/58758556-62529280-854f-11e9-8429-a436c988bbfc.png)

The Ensembl grch38.84 fasta is filtered to retain gene-biotypes as in [this note](https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build#grch38_1.2.0)
as I check the fasta file, mt transcripts are present:
![image](https://user-images.githubusercontent.com/8361080/58758602-0dfbe280-8550-11e9-81b8-c6292d31e821.png)

**Comment by chriscainx** [comment]
Calculation of MT and RP portion:
```{r}
seo[["percent.mt"]] <- PercentageFeatureSet(seo, pattern = "^MT-")
seo[["percent.rp"]] <- PercentageFeatureSet(seo, pattern = "^RP[SL]")
```

**Comment by pmelsted** [comment]
Which workflow are you using to go from the BUS file to a gene count matrix compatible with seurat? Somewhere in the translation the names might be messed up or missing, can you check that the gene symbols are there in the Seurat object and/or in the file that maps ENST ids to genes and gene symbols.

**Comment by gaofan83** [comment]
Did you try the latest bustools? https://bustools.github.io/download
It will be helpful if you can show us the first several lines of tr2g file (needed for bustools), and the output genes.txt file.

**Comment by chriscainx** [comment]
> Which workflow are you using to go from the BUS file to a gene count matrix compatible with seurat? Somewhere in the translation the names might be messed up or missing, can you check that the gene symbols are there in the Seurat object and/or in the file that maps ENST ids to genes and gene symbols.

I used the workflow in the documentation of BUSpaRse. 
```{r}
 make_sparse_matrix("test/run/kaout/output_sorted.txt",
                              tr2g = tr2g, est_ncells = 1000000,
                              est_ngenes = nrow(tr2g),
                              whitelist = whitelist_v2, TCC = FALSE)
```

> Did you try the latest bustools? https://bustools.github.io/download
> It will be helpful if you can show us the first several lines of tr2g file (needed for bustools), and the output genes.txt file.

The head of tr2g is:
```{r}
             gene        transcript    gene_name
1 ENSG00000243485 ENST00000473358.1 RP11-34P13.3
2 ENSG00000243485 ENST00000469289.1 RP11-34P13.3
3 ENSG00000237613 ENST00000417324.1      FAM138A
4 ENSG00000237613 ENST00000461467.1      FAM138A
5 ENSG00000186092 ENST00000335137.3        OR4F5
6 ENSG00000238009 ENST00000466430.5 RP11-34P13.7
```
I paid special attention to the tr2g file and made sure it is generated from the same reference as cellranger, and went through same filtering as cellranger mkref does.

I haven't tried the new version of bustools. Is there a vital change?

**Comment by lambdamoses** [comment]
![image](https://user-images.githubusercontent.com/32186361/60220495-0f58bc00-982b-11e9-9bbd-f3c05ec14598.png)
This is what I got with kallisto and `BUSpaRse` with the pbmc1k v3 chemistry dataset. So I don't think the actual problem comes from kallisto.

**Comment by chriscainx** [comment]
Hi, I re ran the analysis using the latest version of kallisto. looks like this time Mt genes are preserved, but HLA's are eliminated.
![image](https://user-images.githubusercontent.com/8361080/61447821-55b1be80-a984-11e9-917c-a73b0b4ecdb2.png)

![image](https://user-images.githubusercontent.com/8361080/61447787-3d41a400-a984-11e9-9a53-7f74d390b549.png)

**Comment by chriscainx** [comment]
Hi, just writing to ask if you guys have identified the problem...

**Comment by pmelsted** [comment]
Can you grep your cdna fasta file for `ENST00000425848`. This is a transcript for one of the alternative genes for HLA-B. If it's in the index then kallisto will pseudoalign to it, however it's not in the  regular GTF file (you have to use the one that is called `chr_patch_hapl_scaff.gtf` to obtain this information). Presumably what's happening is that reads are aligning to these genes, which have a separate `gene_id` but the same `gene_name` in the GTF and thus `bustools count` treats them as multimapping and discards them.

I think the proper way to fix this is to collapse all the counts by faking the `gene_id` of all the HLA-B genes to be the `gene_id` for the one on `chr6`.

**Comment by chriscainx** [comment]
Yes, this ENST is indeed in the fasta, and not in the gtf (both downloaded from ensembl). I agree this seems to be a gene id problem... It'll be great if you guys can provide a hotfix with bustools!

**Comment by ramit29** [comment]
@chriscainx the potential issue you're facing with the HLA genes is that they are highly polymorphic. In this case if your sample of interest's HLA genes are significantly polymorphic when compared to the reference genome, then there would be no optimal T-DBG path in the index built using the reference transcriptome. A fix might be to replace the HLA region in the reference transcriptome with your sample specific HLA region.
